blog/_posts/2024-06-06-istio-native-sidecar.md

```diff
@@ -0,0 +1,164 @@
+   2024-06-04T05:07:02.138613Z    info    cache    returned workload trust anchor from cache    ttl=23h59m59.861388298s
+   2024-06-04T05:07:10.904779Z    info    Readiness succeeded in 9.015247842s
+   2024-06-04T05:07:10.905752Z    info    Envoy proxy is ready
+   ```
+
+   Events show application container was created 10 seconds after istio-proxy container was created. That is 1 second after the istio-proxy sidecar became ready.
+
+   | LAST SEEN | TYPE | REASON | OBJECT | MESSAGE |
+   |-----------|------|--------|--------|---------|
+   | 2m38s | Normal | Started | pod/httpbin-9c5fdf746-hzmsf | Started container istio-proxy |
+   | 2m38s | Normal | Started | pod/httpbin-9c5fdf746-hzmsf | Started container istio-init |
+   | 2m28s | Normal | Created | pod/httpbin-9c5fdf746-hzmsf | Created container httpbin |
+   | 2m28s | Normal | Started | pod/httpbin-9c5fdf746-hzmsf | Started container httpbin |
+
+   At the termination time, sidecar get killed before the application container.  This gives sidecar time to drain inflight traffic before terminating the application container.
+
+   ```text
+   2024-06-04T05:36:58.999684Z    info    handling /drain, starting drain
+   2024-06-04T05:36:58.999753Z    info    Agent draining proxy
+   2024-06-04T05:36:59.579419Z    info    Status server has successfully terminated
+   2024-06-04T05:36:59.579514Z    info    Agent draining Proxy for termination
+   2024-06-04T05:36:59.579521Z    info    Agent already drained, exiting immediately
+   2024-06-04T05:36:59.579525Z    info    Agent has successfully terminated
+   ```
+   
+   From events, we can see application container is killed after istio-proxy sidecar.
+
+   | LAST SEEN | TYPE | REASON | OBJECT | MESSAGE |
+   |-----------|------|--------|--------|---------|
+   | 2m27s | Normal | Killing | pod/httpbin-598f48c6d-nwx67 | Stopping container istio-proxy |
+   | 2m27s | Normal | SuccessfulDelete | replicaset/httpbin-598f48c6d | Deleted pod: httpbin-598f48c6d-nwx67 |
+   | 2m27s | Normal | Killing | pod/httpbin-598f48c6d-nwx67 | Stopping container httpbin |
+
+## Conclusion
+
+The integration of native sidecar support in Istio 1.20 and its inclusion in the AKS Istio addon marks a significant advancement in simplifying and enhancing service mesh deployment and management. This feature provides substantial benefits, including improved performance, simplified configuration, and operational efficiency. We are committed to providing state-of-the-art features that empower our users to build resilient and scalable applications on AKS.
+
+## Credits
+
+This feature was made possible through the collaborative efforts of the AKS traffic team, the AKS Add-on team, and the Istio community. Special thanks to Niranjan Shankar, Shashank Barsin and all other AKS traffic team members for the collaboration; Robbie Zhang for the help given from AKS CCP and Add-on team; and Brian Redmond for his guidance on this blog and last but not least Paul Yu for the review.
```

Comment:
these are internal (addon team, traffic) team and will not be relevant for users. we can send an internal mail thanking these folks. suggest taking out this section to align with format of other posts on the blog

Reply:
Not quite sure.  At a common practice, most tech blogs from other companies carries acknowledgements section for other teams or team members who helped. That is a great way to express gratitude.